src/App.tsx

```diff
@@ -1,93 +1,267 @@
+          setHasError(true);
+          setErrorType(TypeError.TITLE);
+          setIsAdding(false);
+
+          return;
+        }
+
+        setTimeout(async () => {
+          await addTodos(newTodoTitle, user.id);
+          await loadTodosFromServer();
+        }, 100);
+
+        setHasError(false);
+        setIsAdding(false);
+        setNewTodoTitle('');
+      } catch {
+        setErrorType(TypeError.ADD);
+        setHasError(true);
+      }
+    }
+  };
+
+  const handleToggleTodo = useCallback(
+    async (todoId: number, completed: boolean) => {
+      setSelectedTodoIds(currentTodoIds => [...currentTodoIds, todoId]);
+
+      try {
+        await toggleTodo(todoId, !completed);
+        await loadTodosFromServer();
+
+        setSelectedTodoIds(currentTodoIds => (
+          currentTodoIds.filter(id => id !== todoId)
+        ));
+      } catch {
+        setErrorType(TypeError.UPDATE);
+        setHasError(true);
+        setSelectedTodoIds([]);
+      }
+    }, [],
+  );
+
+  const clearAllHandler = useCallback(async () => {
+    try {
+      const completedTodoIds = todos
+        .filter(todo => todo.completed)
+        .map(todo => todo.id);
+
+      setDeletedTodoIds(completedTodoIds);
+
+      await Promise.all(todos.map(async todo => {
+        if (todo.completed) {
+          await deleteTodos(todo.id);
+        }
+
+        return todo;
+      }));
+
+      await loadTodosFromServer();
+    } catch {
+      setErrorType(TypeError.DELETE);
+      setHasError(true);
+      setDeletedTodoIds([]);
+    }
+  }, [todos]);
+
+  const deleteTodoHandler = useCallback(
+    async (todoId: number) => {
+      setDeletedTodoIds([todoId]);
+
+      try {
+        await deleteTodos(todoId);
+        await loadTodosFromServer();
+      } catch {
+        setErrorType(TypeError.DELETE);
+        setHasError(true);
+        setDeletedTodoIds([]);
+      }
+    }, [],
+  );
+
+  const handleToggleAllTodos = useCallback(async () => {
+    try {
+      await Promise.all(todos.map(async (todo) => {
+        if (!todo.completed || isEachTodoCompleted) {
+          setSelectedTodoIds(currentTodoIds => [...currentTodoIds, todo.id]);
+
+          await toggleTodo(todo.id, !todo.completed);
+        }
+      }));
+
+      await loadTodosFromServer();
+
+      setSelectedTodoIds([]);
+    } catch {
+      setErrorType(TypeError.UPDATE);
+      setHasError(true);
+      setSelectedTodoIds([]);
+    }
+  }, [selectedTodoIds]);
+
+  const handleChangeTodoTittle = useCallback(
+    async (todoId: number, title: string) => {
+      setSelectedTodoIds([todoId]);
+
+      try {
+        await renameTodo(todoId, { title });
+        await loadTodosFromServer();
+
+        setSelectedTodoIds([]);
+      } catch {
+        setErrorType(TypeError.UPDATE);
+        setHasError(true);
+
+        setSelectedTodoIds([]);
+      }
+    }, [],
+  );
+
+  useEffect(() => {
+    loadTodosFromServer();
+  }, [todos]);
```

Comment:
Don't you need to pass the `loadTodosFromServer` function to the array?
```suggestion
  useEffect(() => {
    loadTodosFromServer();
  }, [todos, loadTodosFromServer]);
```